unit_testing/LightController/LightController_tests.cpp

```diff
@@ -410,4 +410,87 @@ TEST_F(LightControllerTest, LEDSwitching)
     goToTick(171);
     for(int i = 0; i < LED_CHANNELS*NO_COLORS - 1; i++) EXPECT_FALSE(stmGetGpio(*ChCtrl[i]));
     EXPECT_TRUE(stmGetGpio(*ChCtrl[11]));
-} 
\ No newline at end of file
+} 
+
+TEST_F(LightControllerTest, LEDSwitchingTimeout)
+{
+    LightControllerInit(&htim2, &htim5, &hwwdg);
+    LightControllerLoop(bootMsg);
+
+    /* --- Turn on white LED for p1-3  --- */
+    writeLightControllerMessage("p1 FFFFFF\n");
+    writeLightControllerMessage("p2 FFFFFF\n");
+    writeLightControllerMessage("p3 FFFFFF\n");
+
+    HAL_TIM_PeriodElapsedCallback(&htim5);
+    EXPECT_FLUSH_USB(ElementsAre(
+        "\r", 
+        "Boot Unit Test\r",
+        "ffffff, ffffff, ffffff, 0x7"
+    )); 
+
+    forceTick(1);                   // Set system tick to 1
+    LightControllerLoop(bootMsg);   // Run main loop which monitors ACTUATION_TIMEOUT
+    goToTick(1);                    // LED switching interrupt
+
+    HAL_TIM_PeriodElapsedCallback(&htim5);
+    EXPECT_FLUSH_USB(ElementsAre(
+        "\r",
+        "ffffff, ffffff, ffffff, 0x7"
+    )); 
+
+    // All white GPIOs should be on
+    EXPECT_TRUE(stmGetGpio(*ChCtrl[3]));
+    EXPECT_TRUE(stmGetGpio(*ChCtrl[7]));
+    EXPECT_TRUE(stmGetGpio(*ChCtrl[11]));
+
+    forceTick(25123);               // Set system tick to 25123 i.e. < ACTUATION_TIMEOUT
+    LightControllerLoop(bootMsg);   // Run main loop which monitors ACTUATION_TIMEOUT
+    goToTick(2);                    // LED switching interrupt
+
+    HAL_TIM_PeriodElapsedCallback(&htim5);
```

Comment:
I can't comment on the actual problematic area of the unit test here. Its not a big deal and its clear the test works, but if simTick was improved slightly, you wouldn't need to:
```
forceTick, 
LiquidControllerLoop
goToTick
HAL_TIM_PeriodElapsed
```
each time you want to make a change.

See DC_tests.cpp for an example